pagefaultgames/pokerogue
# [Bug] Fix eggs having exploitable RNG

## What are the changes the user will see?

Egg RNG manipulation will no longer work.

## Why am I making these changes?



A temporary fix for #3607 before Xavion's planned RNG rework.

## What are the changes from a developer perspective?


Implements 2 fixes at the same time:

1. Properly resets RNG upon saving and quitting into the title screen. This protects against RNG manipulating anything accessible from the title screen, including starter eggs.
    - Also fixes some unit tests that have a new random chance to fail due to these RNG resets.
2. In `egg.ts`, uses `scene.executeWithSeedOffset()` to make egg RNG unseeded and unpredictable. This protects against RNG manipulation for all eggs, including candy eggs and gacha eggs.

### Screenshots/Videos



## How to test the changes?




Here were the steps for reproducing the original bug:

1. Load a session (after saving and quitting at least once).
2. Save and quit (NOT refresh).
3. Navigate to the starter select screen.
4. Buy a single starter egg using candies.
5. Repeat from Step 1 for each egg.

If the fix works properly, then the eggs should have different egg move slots, ability slots, shininess, etc..

## Checklist
- [x] There is no overlap with another PR? (not until Xavion's planned RNG rework)
